include/boost/math/statistics/univariate_statistics.hpp

```diff
@@ -512,6 +512,63 @@ inline auto interquartile_range(RandomAccessContainer & v)
     return interquartile_range(v.begin(), v.end());
 }
 
+template<class ForwardIterator, class OutputIterator>
+auto sorted_mode(ForwardIterator first, ForwardIterator last, OutputIterator output) -> decltype(output)
+{
+    using Z = typename std::iterator_traits<ForwardIterator>::value_type;
+    static_assert(std::is_integral<Z>::value, "Floating point values have not yet been implemented.");
+    using Size = typename std::iterator_traits<ForwardIterator>::difference_type;
+
+    std::vector<Z> modes {};
+    modes.reserve(16);
```

Comment:
A performance idea, and honestly you're under no obligation to pursue it. You can just ignore me. But did you consider trying `boost::container::small_vector` here? Because I assume _most_ of the time, `modes` is not going to grow large at all. Or it might not make a difference. Entirely up to you.

Reply:
@jeremy-murphy : It would couple `boost::container` into `boost::math`.

Reply:
@jeremy-murphy By calling `std::vector::reserve()` and using `std::vector::emplace_back()` dynamic memory allocation is avoided in most cases, which from the docs looks like what `boost::container::small_vector` accomplishes.

Reply:
OK, good, just a thought.  :)